src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -360,6 +389,9 @@ def __new_installation_summary_json(self, installation_packages_json):
 
     def set_patch_metadata_for_healthstore_substatus_json(self, status=Constants.STATUS_SUCCESS, code=0, patch_version=Constants.PATCH_VERSION_UNKNOWN, report_to_healthstore=False, wait_after_update=False):
         """ Prepare the healthstore substatus json including message containing summary to be sent to healthstore """
+        if self.execution_config.exec_auto_assess_only and status == Constants.STATUS_TRANSITIONING:
+            raise Exception("Auto-assessment mode. Unexpected attempt to update healthstore status.")
```

Comment:
@mirichmo / @kjohn-msft For an auto assessment, do we report anything to healthstore? Eg. do we have healthstore substatus in the status blob?

Reply:
In the Windows extension, PatchMetadataForHealthStore is only written during installation operations. I'm unsure about how the Linux extension does it, but I'd assume it is the same.